apps/laboratory/src/layout/CustomWallet.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  Button,
+  Drawer,
+  DrawerBody,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerHeader,
+  DrawerOverlay,
+  Input,
+  Link,
+  Stack,
+  Text,
+  useDisclosure
+} from '@chakra-ui/react'
+import { useState, type ChangeEvent } from 'react'
+import { CUSTOM_WALLET } from '../utils/ConstantsUtil'
+
+interface Props {
+  controls: ReturnType<typeof useDisclosure>
+}
+
+export function CustomWallet({ controls }: Props) {
+  const [customWallet, setCustomWallet] = useState({
+    id: 'custom',
+    name: '',
+    image_url: '',
+    mobile_link: '',
+    desktop_link: '',
+    webapp_link: ''
+  })
+
+  function handleCustomWallet() {
+    localStorage.setItem(CUSTOM_WALLET, JSON.stringify(customWallet))
+    location.reload()
```

Comment:
Why do we need to reload? Is it something we might need to improve on w3m?

Reply:
to re-instantiate Web3Modal